iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/pipe/PipeTaskInfo.java

```diff
@@ -244,6 +255,71 @@ public TSStatus handleMetaChanges(PipeHandleMetaChangePlan plan) {
     return new TSStatus(TSStatusCode.SUCCESS_STATUS.getStatusCode());
   }
 
+  /**
+   * Clear the exceptions of a pipe locally after it starts successfully. If there are exceptions
+   * cleared, the messages will then be updated to all the nodes through
+   * PipeHandleMetaChangeProcedure.
+   *
+   * @param pipeName The name of the pipes to be clear exception
+   * @return true if there are exceptions cleared
+   */
+  public boolean clearExceptions(String pipeName) {
+    AtomicBoolean isCleared = new AtomicBoolean(false);
+    PipeRuntimeMeta runtimeMeta = pipeMetaKeeper.getPipeMeta(pipeName).getRuntimeMeta();
+    runtimeMeta.setExceptionsClearTime(System.currentTimeMillis());
+    Map<Integer, PipeRuntimeException> exceptionMap =
+        runtimeMeta.getDataNodeId2PipeRuntimeExceptionMap();
+    if (!exceptionMap.isEmpty()) {
+      exceptionMap.clear();
+      isCleared.set(true);
+    }
+    runtimeMeta
+        .getConsensusGroupId2TaskMetaMap()
+        .values()
+        .forEach(
+            pipeTaskMeta -> {
+              if (pipeTaskMeta.getExceptionMessages().iterator().hasNext()) {
+                isCleared.set(true);
+                pipeTaskMeta.clearExceptionMessages();
+              }
+            });
+    return isCleared.get();
+  }
+
+  /**
+   * Record the exceptions of all pipes locally if they encountered failure when pushing pipe meta.
+   * If there are exceptions recorded, the related pipes will be stopped, and the exception messages
+   * will then be updated to all the nodes through PipeHandleMetaChangeProcedure.
+   *
+   * @param respMap The responseMap after pushing pipe meta
+   * @return true if there are exceptions encountered
+   */
+  public boolean recordPushPipeMetaExceptions(Map<Integer, TPushPipeMetaResp> respMap) {
+    boolean hasException = false;
+    for (Map.Entry<Integer, TPushPipeMetaResp> respEntry : respMap.entrySet()) {
+      int dataNodeId = respEntry.getKey();
+      TPushPipeMetaResp resp = respEntry.getValue();
+      if (resp.getStatus().getCode() == TSStatusCode.PUSH_PIPE_META_ERROR.getStatusCode()) {
+        hasException = true;
+        resp.getExceptionMessages()
+            .forEach(
+                message -> {
+                  PipeRuntimeMeta runtimeMeta =
+                      pipeMetaKeeper.getPipeMeta(message.getPipeName()).getRuntimeMeta();
+                  // Mark the status of the pipe with exception as stopped
+                  runtimeMeta.getStatus().set(PipeStatus.STOPPED);
+                  runtimeMeta
+                      .getDataNodeId2PipeRuntimeExceptionMap()
+                      .put(
+                          dataNodeId,
+                          new PipeRuntimeCriticalException(
+                              message.getMessage(), message.getTimeStamp()));
```

Comment:
if there is already an exception recorded.. should we judge the time and record the latest one?